sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/externalrules/CxxExternalRulesSensor.java

```diff
@@ -59,11 +72,30 @@ protected String reportPathKey() {
   public void describe(SensorDescriptor descriptor) {
     descriptor.onlyOnLanguage(CxxLanguage.KEY).name("CxxExternalRulesSensor");
   }
-  
+
+  /**
+   * {@inheritDoc}
+   */
   @Override
-  public void processReport(final SensorContext context, File report) throws XMLStreamException {
+  public void execute(SensorContext context) {
+    try {
+      transformFiles();
+    } catch (Exception e) {
```

Comment:
Doing so would then result in a lot of errors if one transformation fails, because the source of the next transformation step would be missing. I know, that this would be more tolerant.

Its a design decision, which I would leave up to you. So if you want to go on transforming files, even there are errors, then I would implement it that way.